frontend_tests/node_tests/markdown.js

```diff
@@ -291,6 +291,12 @@ var bugdown_data = JSON.parse(fs.readFileSync(path.join(__dirname, '../../zerver
          expected: '<p>@*notagroup*</p>'},
         {input: 'This is a realm filter `hello` with text after it',
          expected: '<p>This is a realm filter <code>hello</code> with text after it</p>'},
+        // Test the emoticon conversion
+        {input: ':)',
+         expected: '<p>:)</p>'},
+        {input: ':)',
+         expected: '<p><span class="emoji emoji-1f603" title="smiley">:smiley:</span></p>',
+         translate_emoticons: true},
```

Comment:
This is better than before, but even better would be to have the `translate_emoticons` test case in the shared suite between backend and frontend (`zerver/fixtures/markdown_test_cases.json`) support the `translate_emoticons` key.